### PR TITLE
CA1851: Possible multiple enumerations of IEnumerable collection

### PR DESCRIPTION
**Pull request by v-ajajvanu**
## Description
userstory 100880 part 2 changes

## Related issues
user story [AB#100880](https://microsofthealth.visualstudio.com/f8da5110-49b1-4e9f-9022-2f58b6124ff9/_workitems/edit/100880)

## Testing
ran unit tests.

## FHIR Team Checklist
- **Update the title** of the PR to be succinct and less than 65 characters
- **Add a milestone** to the PR for the sprint that it is merged (i.e. add S47)
- Tag the PR with the type of update: **Bug**, **Build**, **Dependencies**, **Enhancement**, **New-Feature** or **Documentation**
- Tag the PR with **Open source**, **Azure API for FHIR** (CosmosDB or common code) or **Azure Healthcare APIs** (SQL or common code) to specify where this change is intended to be released.
- Tag the PR with **Schema Version backward compatible** or **Schema Version backward incompatible** or **Schema Version unchanged** if this adds or updates Sql script which is/is not backward compatible with the code.
- [ ] CI is green before merge [Build Status](https://microsofthealthoss.visualstudio.com/FhirServer/_build/latest?definitionId=27&branchName=main) 
- Review [squash-merge requirements](https://github.com/microsoft/fhir-server/blob/master/SquashMergeRequirements.md)

### Semver Change ([docs](https://github.com/microsoft/fhir-server/blob/master/docs/Versioning.md))
Patch|Skip|Feature|Breaking (reason)

### CODE REVIEW

**Review by abiisnn — COMMENTED** [review]
/azp run

**Review by abiisnn — COMMENTED** [review]
/azp run

File: src/Microsoft.Health.Fhir.Shared.Core/Features/Search/SearchOptionsFactory.cs [hunk 1]
```diff
@@ -529,7 +529,7 @@ private IEnumerable<IncludeExpression> ParseIncludeIterateExpressions(IList<(str
                     allowedResourceTypesByScope = _contextAccessor.RequestContext?.AccessControlContext?.AllowedResourceActions.Select(s => s.Resource);
                 }
 
-                var expression = _expressionParser.ParseInclude(includeResourceTypeList, p.query, isReversed, iterate, allowedResourceTypesByScope);
+                var expression = _expressionParser.ParseInclude(includeResourceTypeList, p.query, isReversed, iterate, allowedResourceTypesByScope.ToList());
```
Comment (LTA-Thinking) [inline src/Microsoft.Health.Fhir.Shared.Core/Features/Search/SearchOptionsFactory.cs:532]: ToList can't be used here without a null check. This is probably causing some of the test failures.

**Review by LTA-Thinking — APPROVED** [review]

**Review by LTA-Thinking — APPROVED** [review]

**Comment by rbans96** [comment]
@v-ajajvanu Service is failing to come up due to DI issue. Here is the [wiki ](https://dev.azure.com/microsofthealth/Health/_wiki/wikis/Health.wiki/1158/PR-Build-and-Deploy-issues?anchor=1.-response-status-code-does-not-indicate-success%3A-503-(service-unavailable))page for the details how to find this log.

2024_07_30_lw1ldlwk000018_default_docker.log:
2024-07-30T09:55:57.419196488Z Application startup exception: System.InvalidOperationException: Unable to resolve service for type 'System.Collections.Generic.IReadOnlyCollection`1[Microsoft.Health.Fhir.Core.Features.Search.Converters.ITypedElementToSearchValueConverter]' while attempting to activate 'Microsoft.Health.Fhir.Core.Features.Search.Converters.FhirTypedElementToSearchValueConverterManager'.
2024-07-30T09:55:57.419261589Z at

**Review by fhibf — COMMENTED** [review]
Please, rename this PR. Just from the current title it's not possible to understand what it's being changed.

**Comment by v-ajajvanu** [comment]
Closing this PR as changes here are no longer needed ad are covered in separate PR

**Pull request closed without merging**